pkg/controller/jobs/job/job_webhook_test.go

```diff
@@ -188,6 +193,19 @@ func TestValidateCreate(t *testing.T) {
 				field.Invalid(syncCompletionAnnotationsPath, "true", "should not be enabled for NonIndexed jobs"),
 			},
 		},
+		{
+			name: "valid sync completions annotation, server version less then 1.27",
+			job: testingutil.MakeJob("job", "default").
+				Parallelism(4).
+				Completions(4).
+				SetAnnotation(JobCompletionsEqualParallelismAnnotation, "true").
+				Indexed(true).
+				Obj(),
+			wantErr: field.ErrorList{
+				field.Invalid(syncCompletionAnnotationsPath, "true", "only supported in Kubernetes 1.27 or newer"),
+			},
+			serverVersion: "1.26.3",
+		},
 		{
 			name: "valid sync completions annotation",
```

Comment:
why is this test passing if you didn't set a version?

Reply:
it shouldn't, I added test case where we don't specify server version